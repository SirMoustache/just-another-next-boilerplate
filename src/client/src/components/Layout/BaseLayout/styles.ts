/**
 * Absolute imports
 */
import styled from 'styled-components';

export const Root = styled.div`
  display: grid;
  grid-template-rows: auto 1fr auto;
  min-height: 100vh;
`;
