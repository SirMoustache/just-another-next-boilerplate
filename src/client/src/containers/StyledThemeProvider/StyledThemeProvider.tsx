/**
 * Absolute imports
 */
import React, { FC, Fragment } from 'react';
import { ThemeProvider } from 'styled-components';

/**
 * Theme
 */
import lightTheme from '../../theme/light';

export type StyledThemeProviderProps = {
  theme?: 'light';
};

const themes = {
  light: lightTheme,
};

const StyledThemeProvider: FC<StyledThemeProviderProps> = ({
  children,
  theme = 'light',
}) => {
  const activeTheme = themes[theme];
  return (
    <ThemeProvider theme={activeTheme}>
      <Fragment>{children}</Fragment>
    </ThemeProvider>
  );
};

export default StyledThemeProvider;
