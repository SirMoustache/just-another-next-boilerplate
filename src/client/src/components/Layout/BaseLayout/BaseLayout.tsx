/**
 * Absolute imports
 */
import React, { Fragment, FC } from 'react';
import { motion } from 'framer-motion';

/**
 * Components
 */
import Header from '../Header';
import Footer from '../Footer';

/**
 * Styles
 */
import { Root } from './styles';

export type BaseLayoutProps = {
  justify?: 'center' | 'flex-end' | 'flex-start';
  align?: 'center' | 'flex-end' | 'flex-start';
};

const BaseLayout: FC = ({ children }) => {
  return (
    <Root>
      <Header />
      <motion.main initial="initial" animate="animate" exit={{ opacity: 0 }}>
        {children}
      </motion.main>
      <Footer />
    </Root>
  );
};

export default BaseLayout;
