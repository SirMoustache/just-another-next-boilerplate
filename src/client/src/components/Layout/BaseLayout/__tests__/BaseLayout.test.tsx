/**
 * Absolute imports
 */
import React from 'react';
import { render, cleanup } from '@testing-library/react';

/**
 * Containers
 */
import StyledThemeProvider from '../../../../containers/StyledThemeProvider';

/**
 * Components
 */
import BaseLayout from '../BaseLayout';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

const children = <span data-test="child">Test</span>;
const renderComponent = (props = {}) =>
  render(
    <StyledThemeProvider>
      <BaseLayout {...props}>{children}</BaseLayout>
    </StyledThemeProvider>,
  );

describe('<BaseLayout />', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render a <div> tag', () => {
    const { container } = renderComponent();
    expect(container.querySelector('div')).not.toBeNull();
  });

  it('should have render children', () => {
    const { container } = renderComponent();
    // @ts-ignore
    expect(container.querySelector('[data-test="child"]')).not.toBeNull();
  });
});
