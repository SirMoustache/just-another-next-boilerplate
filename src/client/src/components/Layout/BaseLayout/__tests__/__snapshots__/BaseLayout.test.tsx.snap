// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseLayout /> should match snapshot 1`] = `
<div>
  <div
    class="sc-jzJRlG faDjRE"
  >
    <header
      class="sc-ifAKCX cwJIEf"
    >
      <div
        class="sc-EHOje kLLRRQ"
      >
        <div
          class="sc-bZQynM jdkaLI"
        >
          <img
            alt="Logo"
            class="sc-gzVnrw giKwVo"
            src=""
          />
        </div>
        <nav
          class="sc-bdVaJa bmJbVs"
        >
          <a
            class="sc-bwzfXH ghJrjT"
            href="/"
          >
            Home
          </a>
          <a
            class="sc-bwzfXH ghJrjT"
            href="/about"
          >
            About
          </a>
          <a
            class="sc-bwzfXH ghJrjT"
            href="/contact"
          >
            Contact
          </a>
        </nav>
        <div
          class="sc-htoDjs jWUwCq"
        >
          <div
            class="sc-iwsKbI haoHLS"
          >
            <div
              class="sc-htpNat cdeDWD"
            >
              <span
                class="sc-bxivhb ePlLNy"
              />
              <span
                class="sc-bxivhb ePlLNy"
              />
              <span
                class="sc-bxivhb ePlLNy"
              />
            </div>
          </div>
        </div>
      </div>
    </header>
    <span
      data-test="child"
    >
      Test
    </span>
    <footer
      class="sc-gZMcBi gfBlKl"
    >
      <div
        class="sc-gqjmRU ljvdtu"
      >
        <div
          class="sc-VigVT fjkIrR"
        >
          <h4
            class="sc-jTzLTM gWHzEN"
          >
            Products
          </h4>
          <div
            class="sc-fjdhpX bPqjga"
          >
            Random Item
          </div>
          <div
            class="sc-fjdhpX bPqjga"
          >
            More Items
          </div>
          <div
            class="sc-fjdhpX bPqjga"
          >
            Even More Items
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
